compiler/CodeGenVisitor.cpp

```diff
@@ -42,52 +44,99 @@ antlrcpp::Any CodeGenVisitor::visitInit(ifccParser::InitContext *ctx)
-
-	if( vars.find(varname)==vars.end()){
-		this->vars[varname]=index;
-		cout << "\tmovl " + value + ", -" + to_string(index) + "(%rbp)" << endl;
-	  return 0;
-	}else{
-		error=true;
-		return 0;
+	if (vars.find(varname) == vars.end()) {
+		this->vars[varname] = index;
+		cout << "\tmovl " + value + ", " << EAX << endl;
+	  cout << "\tmovl " + EAX + ", -" + to_string(index) + "(%rbp)" << endl;
+	} else {
+		error = true;
 	}
-	
-	
-
-	
-
+	return 0;
 }
 
 antlrcpp::Any CodeGenVisitor::visitValue(ifccParser::ValueContext *ctx) 
 {
 	string returnval;
 	antlr4::tree::TerminalNode * varnameNode = ctx->VARNAME();
 	if (varnameNode) {
-		std::string varname = varnameNode->getText();
-		std::string index = std::to_string(this->vars[varname]);
+		string varname = varnameNode->getText();
+		string index = to_string(this->vars[varname]);
 		returnval = "-" + index + "(%rbp)";
 	} else {
 		returnval = "$" + ctx->CONST()->getText();
 	}
 	return returnval;
 }
 
+string CodeGenVisitor::operationExpression(string leftval, string rightval, string operation) {
+	int index = (this->vars.size() + 1) * 8;
+	string indexString = to_string(index);
+	string varname = "temp" + indexString;
+	this->vars[varname] = index;
+	std::string regval = "-" + indexString + "(%rbp)";
+	cout << "\tmovl " << leftval << ", " << EAX << endl;
+  cout << "\t" << operation << " " << rightval << ", " << EAX << endl;
+	cout << "\tmovl " << EAX << ", " << regval << endl;
+	return regval;
+}
+
 antlrcpp::Any CodeGenVisitor::visitExpressionMult(ifccParser::ExpressionMultContext *ctx) 
 {
 	string leftval = visit(ctx->expression(0)).as<string>();
-	cout << "\tmovl " << leftval << ", %eax" << endl;
 	string rightval = visit(ctx->expression(1)).as<string>();
-  cout << "\timull " << rightval << ", %eax" << endl;
-	return string("%eax");
+	return operationExpression(leftval, rightval, "imull");
+}
+
+antlrcpp::Any CodeGenVisitor::visitExpressionDiv(ifccParser::ExpressionDivContext *ctx) 
+{
+	string leftval = visit(ctx->expression(0)).as<string>();
+	string rightval = visit(ctx->expression(1)).as<string>();
+	int index = (this->vars.size() + 1) * 8;
+	string indexString = to_string(index);
+	string varname = "temp" + indexString;
+	this->vars[varname] = index;
+	std::string regval = "-" + indexString + "(%rbp)";
+	cout << "\tmovl " << leftval << ", " << EAX << endl;
+	cout << "\tcltd" << endl;
+  cout << "\tmovl " << rightval << ", " << ECX << endl;
+	cout << "\tidivl " << ECX << endl;
+	cout << "\tmovl " << EAX << ", " << regval << endl;
+	return regval;
 }
 
 antlrcpp::Any CodeGenVisitor::visitExpressionAdd(ifccParser::ExpressionAddContext *ctx) 
 {
 	string leftval = visit(ctx->expression(0)).as<string>();
-	cout << "\tmovl " << leftval << ", %eax" << endl;
 	string rightval = visit(ctx->expression(1)).as<string>();
-  cout << "\tiaddl " << rightval << ", %eax" << endl;
-	return string("%eax");
+	return operationExpression(leftval, rightval, "add");
```

Comment:
perhaps instead of p[utting strings it would be nice to use constants at the top 
```cpp
const string ADD_OP =  "add"
```
same goes for other operations